packages/zowe-explorer/src/commands/UnixCommandHandler.ts

```diff
@@ -271,115 +228,18 @@ export class UnixCommandHandler extends ZoweCommandProvider {
-            const choice = await Gui.resolveQuickPick(quickpick);
-            quickpick.hide();
-            if (!choice) {
-                this.serviceProf = undefined;
-                ZoweLogger.info(this.unixCmdMsgs.opCancelledMsg);
-                Gui.showMessage(this.unixCmdMsgs.opCancelledMsg);
-                return;
-            }
-            if (choice instanceof FilterDescriptor) {
-                if (quickpick.value) {
-                    response = quickpick.value;
-                }
-            } else {
-                response = choice.label;
-            }
-        }
-        if (!response || alwaysEdit) {
-            const options2: vscode.InputBoxOptions = {
-                prompt: vscode.l10n.t("Enter or update the Unix command"),
-                value: response,
-                valueSelection: response ? [response.length, response.length] : undefined,
-            };
-            // get user input
-            response = await Gui.showInputBox(options2);
-            if (!response) {
-                this.serviceProf = undefined;
-                ZoweLogger.info(this.unixCmdMsgs.opCancelledMsg);
-                Gui.showMessage(this.unixCmdMsgs.opCancelledMsg);
-                return;
-            }
-        }
-        return response;
-    }
-
-    private async issueCommand(profile: imperative.IProfileLoaded, command: string, cwd: string): Promise<void> {
-        ZoweLogger.trace("UnixCommandHandler.issueCommand called.");
-        const profName = this.sshProfile !== undefined ? this.sshProfile.name : profile.name;
-        try {
-            if (command) {
-                const user: string = profile.profile.user;
-                if (this.sshProfile) {
-                    this.outputChannel.appendLine(`> ${user}@${this.sshProfile.name}:${cwd ? cwd : "~"}$ ${command}`);
-                } else {
-                    this.outputChannel.appendLine(`> ${user}:${cwd ? cwd : "~"}$ ${command}`);
-                }
-                const submitResponse = await Gui.withProgress(
-                    {
-                        location: vscode.ProgressLocation.Notification,
-                        title: vscode.l10n.t("Unix command submitted."),
-                    },
-                    () => {
-                        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-                        return ZoweExplorerApiRegister.getCommandApi(profile).issueUnixCommand(command, cwd, this.sshSession);
-                    }
-                );
-                this.outputChannel.appendLine(submitResponse);
-                this.outputChannel.show(true);
-                this.history.addSearchHistory(command);
-                this.serviceProf = undefined;
-            }
-        } catch (error) {
-            await AuthUtils.errorHandling(error, profName);
-        }
+    public runCommand(profile: imperative.IProfileLoaded, command: string): Promise<string> {
+        this.nodeProfile = undefined;
```

Comment:
Shouldn't this be `this.nodeProfile = profile`? Feel free to correct me if I'm missing something 😅

Reply:
I believe there was an edge case for extenders running SSH commands without an explicit SSH profile, but I don't remember tbh...
Removing it doesn't have any effect since the provided profile is always used as a backup 😅